lib/param_parsing/environment_test.go

```diff
@@ -97,3 +101,83 @@ func TestGetParamsFromEnvironment_log_level_from_env(t *testing.T) {
 		t.Errorf("Determined log level is not matching. Got %q, expected %q", params.LogLevel, expected)
 	}
 }
+
+func TestNoColorEnvVar(t *testing.T) {
+	envVarName := "NO_COLOR"
+	_ = os.Setenv(envVarName, "true")
+	goCommandArgs := []string{"run", "../../main.go", "--dry-run", "1.10.5"}
+
+	t.Logf("Testing %q env var", envVarName)
+
+	out, err := exec.Command("go", goCommandArgs...).CombinedOutput()
+	if err != nil {
+		t.Fatalf("Unexpected failure: \"%v\", output: %q", err, string(out))
+	}
+
+	_ = os.Unsetenv(envVarName)
+
+	matched, err := regexp.MatchString(ansiCodesRegex, string(out))
+	if err != nil {
+		t.Fatalf("Unexpected failure: \"%v\", output: %q", err, string(out))
+	}
+
+	if matched {
+		t.Errorf("Expected no ANSI color codes in output, but found some: %q", string(out))
+	} else {
+		t.Log("Success: no ANSI color codes in output")
+	}
+}
+
+func TestForceColorEnvVar(t *testing.T) {
+	envVarName := "FORCE_COLOR"
+	if color.SupportColor() {
+		_ = os.Setenv(envVarName, "true")
+		goCommandArgs := []string{"run", "../../main.go", "--dry-run", "1.10.5"}
```

Comment:
Is there anything more simple that could be run? It'll be one of those things that whenever there's a test failure for a chunk of the "main" functionality, the color logging tests will also fail

Reply:
> Is there anything more simple that could be run?

Not sure I can imagine anything that would be simpler 🤷🏻 

> It'll be one of those things that whenever there's a test failure for a chunk of the "main" functionality, the color logging tests will also fail

Hopefully the error logged by this test in such a case, would give a hint that "some other" test is the issue 🤞🏻

Reply:
👍